### PR TITLE
Droth 4 paving from vvh

### PR DESCRIPTION
**Pull request by leirasf**

### CODE REVIEW

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/LinearAssetService.scala [hunk 1]
```diff
@@ -96,23 +98,62 @@ trait LinearAssetOperations {
...
+    val filledNewAssets = fillNewRoadLinksWithPreviousAssetsData(projectableTargetRoadLinks,
+      existingAssets.filterNot(a => expiredPavingAssetIds.exists(_ == a.id)) ++ newPavingAssets, assetsOnChangedLinks, changes)
+
+    val newAssets = newPavingAssets.filterNot(a => filledNewAssets.exists(_ == a.linkId)) ++ filledNewAssets
+
     if (newAssets.nonEmpty) {
       logger.info("Transferred %d assets in %d ms ".format(newAssets.length, System.currentTimeMillis - timing))
     }
-    val groupedAssets = (existingAssets.filterNot(a => newAssets.exists(_.linkId == a.linkId)) ++ newAssets).groupBy(_.linkId)
+    val groupedAssets = (existingAssets.filterNot(a => expiredPavingAssetIds.exists(_ == a.id)).filterNot(a => newAssets.exists(_.linkId == a.linkId)) ++ newAssets).groupBy(_.linkId)
```
Comment (tvenhola) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/LinearAssetService.scala]: I am uncertain if this method works with all the cases when assets are transferred and pavement information is updated.

**Comment by marquesfm** [comment]
Eliminate commit at line 1068, file LinearAssetServiceSpec.
Eliminate a ignore case existing at the line 1296.
Commit Done for this modifications
The rest of the code seems well

**Comment by tvenhola** [comment]
Use cases we want to cover:

1. There is no pavement information in OTH entered
- VVH updates go to OTH database as is
2. User has entered pavement information and VVH data is updated later
- VVH timestamp is written to track order of changes
3. VVH has given pavement information which the user later corrects.
- VVH data must not affect the given data
- Later VVH corrections must affect given data

**Comment by tvenhola** [comment]
Looks good to me

**Pull request merged**